plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/TestBloomFilter.java

```diff
@@ -0,0 +1,167 @@
+                assertTrue(bloomFilterOptional.isPresent());
+                BloomFilter bloomfilter = bloomFilterOptional.get();
+
+                for (String testName : TEST_VARCHAR_VALUES) {
+                    Slice testSlice = Slices.utf8Slice(testName);
+                    long testHash = getBloomFilterHash(bloomfilter, testSlice, createVarcharType(255));
+                    assertTrue(bloomfilter.findHash(testHash));
+                }
+
+                Slice testSlice = Slices.utf8Slice("not_exist");
+                long testHash = getBloomFilterHash(bloomfilter, testSlice, createVarcharType(255));
+                assertFalse(bloomfilter.findHash(testHash));
+            }
+            catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
+    public void testReadBloomFilterParquetNonExist()
+            throws Exception
+    {
+        testParquetBloomFilterVarchar(false, parquetFile -> {
+            try {
+                TrinoFileSystemFactory fileSystemFactory = new HdfsFileSystemFactory(HDFS_ENVIRONMENT);
+                TrinoFileSystem fileSystem = fileSystemFactory.create(SESSION);
+                TrinoInputFile inputFile = fileSystem.newInputFile(parquetFile.getPath());
+                TrinoParquetDataSource dataSource = new TrinoParquetDataSource(inputFile, new ParquetReaderOptions(), new FileFormatDataSourceStats());
+
+                ParquetMetadata parquetMetadata = MetadataReader.readFooter(dataSource, Optional.empty());
+                ColumnChunkMetaData columnChunkMetaData = getOnlyElement(getOnlyElement(parquetMetadata.getBlocks()).getColumns());
+                BloomFilterStore bloomFilterStore = new BloomFilterStore(dataSource, getOnlyElement(parquetMetadata.getBlocks()), Set.of(columnChunkMetaData.getPath()));
+                Optional<BloomFilter> bloomFilterOptional = bloomFilterStore.readBloomFilter(columnChunkMetaData.getPath());
+                assertFalse(bloomFilterOptional.isPresent());
+            }
+            catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    private static long getBloomFilterHash(BloomFilter bloomFilter, Slice predicateValue, Type sqlType)
+            throws UnsupportedOperationException
+    {
+        if (sqlType instanceof VarcharType) {
+            return bloomFilter.hash(Binary.fromConstantByteBuffer(predicateValue.toByteBuffer()));
+        }
+        else {
+            throw new UnsupportedOperationException("Unsupported type " + sqlType);
+        }
+    }
+
+    private void testParquetBloomFilterVarchar(boolean enableBloomFilter, Consumer<File> testFunction)
+            throws Exception
+    {
+        MessageType parquetSchema = parseMessageType("message hive_bloomfilter { required binary column1Str (STRING); }");
```

Comment:
I think this can be passed as `Optional.empty`

Reply:
```
  if (schema.isPresent()) {
            DataWritableWriteSupport.setSchema(schema.get(), jobConf);
            return getParquerRecordWriterWrapper(realOutputFormat, jobConf, finalOutPath.toString(), progress, tableProperties);
        }
        return super.getHiveRecordWriter(jobConf, finalOutPath, valueClass, isCompressed, tableProperties, progress);

```

Let me try..

Reply:
done.